### PR TITLE
App refresh bug fix

### PR DESCRIPTION
**Pull request by ItisEeron**
### White Spaces Now Allowed Before and After Email Input
This was a simple fix. I added an optional trim param to the input forms. This is set false by default, so all other input forms not using this will remain virtually the same. 

### All data from the Redux Store is now persists 
I added a new dependency called redux-persists which takes care of the hard work of storing all the data from the redux-store into local/async storage. Now apps should be able to refresh and not have a state clear, they should simply 'rehydrate' to the previous state. 

### Mobile Refreshes Upon Updating 
This involves more redux. I needed to store **appliances of the selected property** into the redux store to get proper behavior of refreshing. I moved some more files around in order to get this to work. All of the modals that mutated properties are now Smart Components (those that connect to the redux store). They use a dispatch method to refresh updates retrieved from the backend. This way, we only need to call dispatch methods to refresh when want too. 

***The modals that edit a property have now been moved into a subdirectory of modals called _property-modals_. It even has its own index files. The base components, types, and functions should still be available for export but I do not include them in the index file.

### CODE REVIEW

File: homepairs/homepairsUI/App.tsx [hunk 1]
```diff
@@ -20,27 +17,9 @@ import store from './src/state/store';
...
-        store.dispatch(refreshServiceProviders(parsedProviders as ServiceProvider[]));
-    }).catch(() => {});
-    
-    await AsyncStorage.getItem('selectedProperty').then(selectedPropertyId => {
-        const storedSelectedPropertyId = selectedPropertyId;
-        store.dispatch(setSelectedProperty(storedSelectedPropertyId));
-    }).catch(() => {});
-    */
+    await AsyncStorage.getItem('persist:root').then(response => {
+        console.log(JSON.parse(response));
```
Comment (YellowRainBoots) [inline homepairs/homepairsUI/App.tsx:21]: let's clean these up next sprint

File: homepairs/homepairsUI/src/app-navigators/AppNavigation.web.tsx [hunk 2]
```diff
@@ -34,6 +34,8 @@ import {
  * 
  * NOTE: Remove all set all flex grow properties to null. This will cause the modal to grow the size of the background 
  * which is unwarranted behavior!!!
+ * 
+ * TODO: Make it so modal is only scrollable, not the background component
```
Comment (YellowRainBoots) [inline homepairs/homepairsUI/src/app-navigators/AppNavigation.web.tsx:38]: should this comment be deleted / isn't the scroll already added below?

File: homepairs/homepairsUI/src/elements/Forms/InputForm.tsx [hunk 3]
```diff
@@ -88,8 +89,8 @@ export default class InputForm extends React.Component<InputFormProps, InputForm
     }
 
     passInputValue(text: string): void {
-        const {parentCallBack} = this.props;
-        parentCallBack(text);
+        const {parentCallBack, trim} = this.props;
+        parentCallBack( trim ? text.trim() : text);
```
Comment (YellowRainBoots) [inline homepairs/homepairsUI/src/elements/Forms/InputForm.tsx:93]: saw this change in the other pull request. hopefully merge will be okay....

File: homepairs/homepairsUI/src/endpoints/api-request.tsx [hunk 4]
```diff
@@ -166,8 +165,6 @@ export const postPreferredProvider = async (
         if(status !== SUCCESS){
             const {error} = data;
             onError(error);
-            console.log(pmId);
-            console.log(error.message);
```
Comment (YellowRainBoots) [inline homepairs/homepairsUI/src/endpoints/api-request.tsx:170]: nice console.log cleaning!

**Comment by ItisEeron** [comment]
It honestly could, I would hold off on merging the branch until I've gotten a chance to test this with the merged code.

File: homepairs/homepairsUI/App.tsx (hunk 1 above)
Comment (ItisEeron) [reply homepairs/homepairsUI/App.tsx:21]: Sounds like a plan. We can definitely have the main focus of the next sprint for clearing up extraneous comments, console logs, and for documentation.

**Pull request merged**